### PR TITLE
[Cache] Add types to private properties

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/Cache/Traits/ContractsTrait.php [hunk 1]
```diff
@@ -83,7 +87,7 @@ static function (CacheItem $item, float $startTime, ?array &$metadata) {
             $startTime = microtime(true);
 
             try {
-                $value = ($this->callbackWrapper)($callback, $item, $save, $pool, function (CacheItem $item) use ($setMetadata, $startTime, &$metadata) {
+                $value = ($this->callbackWrapper ??= \Closure::fromCallable([LockRegistry::class, 'compute']))($callback, $item, $save, $pool, function (CacheItem $item) use ($setMetadata, $startTime, &$metadata) {
```
Comment (stof) [inline src/Symfony/Component/Cache/Traits/ContractsTrait.php]: I would keep the assignment in a separate statement for readability

**Comment by stof** [comment]
Why does Psalm report that `PdoAdapter::doSave` returns null in some cases ?

**Comment by nicolas-grekas** [comment]
> Why does Psalm report that PdoAdapter::doSave returns null in some cases ?

because it doesn't know that the code is always going to enter a foreach.

File: src/Symfony/Component/Cache/Adapter/ProxyAdapter.php [hunk 2]
```diff
@@ -28,18 +28,18 @@ class ProxyAdapter implements AdapterInterface, CacheInterface, PruneableInterfa
...
 
-    private static $createCacheItem;
-    private static $setInnerItem;
+    private static \Closure $createCacheItem;
+    private static \Closure $setInnerItem;
 
     public function __construct(CacheItemPoolInterface $pool, string $namespace = '', int $defaultLifetime = 0)
     {
         $this->pool = $pool;
-        $this->poolHash = $poolHash = spl_object_hash($pool);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/ProxyAdapter.php:42]: the variable is used L50, isn't it?

Comment (nicolas-grekas) [reply src/Symfony/Component/Cache/Adapter/ProxyAdapter.php:42]: OH, no, nvm :)

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**